spec/virtual_includes_spec.rb

```diff
@@ -0,0 +1,180 @@
+describe ActiveRecord::VirtualAttributes::VirtualIncludes do
+  before do
+    Author.destroy_all
+    Book.destroy_all
+    Author.create_with_books(3).books.first.create_bookmarks(2)
+  end
+
+  # books.map(&:author_name)
+  let(:foos) { %w[foo foo foo] }
```

Comment:
Can we call this `author_names` or something?  `foos` isn't very descriptive, and I feel like I would always have to look this up when viewing this in the future.

Reply:
This might need to be another commit, but you could also consider doing in `Author`:

```diff
- def self.create_with_books(count, name=foo)
-   create!(:name => "foo").tap { |author| author.create_books(count) }
+ def self.create_with_books(count, name=foo)
+   create!(:name => name).tap { |author| author.create_books(count) }
```

Which would allow you to change the about line to something like:

```ruby
Author.create_with_books(3, "Keenan").books.first.create_bookmarks(2)
```

And the `let(:foos)` to:

```ruby
let(:author_names) { %w[Keenan Keenan Keenan] }
```

Though, might want to choose a different name if you don't want to look too narcissistic...

Reply:
I fixed `author_name` - it reads much better.
With the change to the way the array works, it is much more straightforward.

I avoided making a real factory for the specs. they should not live in the model but unlike traditional projects, the models are 100% for specs.